### PR TITLE
Add Expect.all

### PR DESCRIPTION
**Pull request by rtfeldman**
Fixes https://github.com/elm-community/elm-test/issues/49

### CODE REVIEW

**Comment by mgold** [comment]
There's another implementation on the `great-expectations` branch. I like that you've done it with explicit recursion. However, there was agreement in #49 that `Expect.all []` should fail.

**Comment by rtfeldman** [comment]
I dig that reasoning. 👍 

Updated to fail on `[]`

File: src/Expect.elm [hunk 1]
```diff
@@ -594,6 +594,9 @@ compareWith label compare expected actual =
 
 {-| Passes if each of the given functions passes when applied to the subject.
 
+**NOTE:** Passing an empty list is assumed to be a mistake, so `Expect.all []`
+will always return a failed expectation no mater what else it is passed.
```
Comment (mgold) [inline src/Expect.elm]: Typo: "mater"

**Comment by mgold** [comment]
No GitHub I don't want to start a review, just leave a comment :ugh:

**Comment by rtfeldman** [comment]
Ha, thanks! Fixed.

**Pull request merged**